projects/components/src/stepper/stepper.component.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  EventEmitter,
+  Input,
+  Output,
+  QueryList
+} from '@angular/core';
+import { MatStepper } from '@angular/material/stepper';
+import { queryListAndChanges$ } from '@hypertrace/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ButtonRole, ButtonStyle } from '../button/button';
+import { IconSize } from '../icon/icon-size';
+import { StepperTabComponent } from './stepper-tab.component';
+
+@Component({
+  selector: 'ht-stepper',
+  template: `
+    <div class="stepper" *ngIf="this.steps$ | async as steps">
+      <mat-stepper [orientation]="this.orientation" [linear]="this.isLinear" #stepper>
+        <ng-container *ngFor="let step of steps; index as i">
+          <mat-step [completed]="step.completed" [stepControl]="step.stepControl" [optional]="step.optional">
+            <ng-template matStepLabel>
+              <ht-label class="header-label" [label]="step.label"></ht-label>
+            </ng-template>
+            <ng-container *ngTemplateOutlet="step.content"></ng-container>
+            <ng-template matStepperIcon="edit">
+              <ht-icon [icon]="step.icon" [size]="'${IconSize.ExtraSmall}'"> </ht-icon>
```

Comment:
nit: you can remove size's '[ ]' and '' from the input, the same for similar variables

Reply:
updated